app/models/notification_group.rb

```diff
@@ -0,0 +1,44 @@
+class NotificationGroup
+  attr_reader :count, :notifications_hash
```

Comment:
```suggestion
  include Enumerable
  attr_reader :notifications_hash
```

We should then get all collections based methods because you defined `each`. We'll get `count`, `map`, `inject` etc for free if we want/need them. It's only a benefit if we want to treat the group like a true collection. Not needed if count, and each is completely enough.

Reply:
Good suggestion, but `count` and `each` are the only ones we need at the moment. I'm happy to revisit this should we want to start treating this class as an Enumerable beyond the `each` method.

Reply:
I like this suggestion

Reply:
When I tested this turns out we still need to have `attr_reader :count`. Otherwise, it will always return `1`, so I'm not sure if we're winning a lot by including `Enumerable` at the moment.